tests/system/common.py

```diff
@@ -226,38 +225,26 @@ def systemctl_master(command='restart'):
     shakedown.run_command_on_master('sudo systemctl {} dcos-mesos-master'.format(command))
 
 
-def save_iptables(host, filename='iptables.rules'):
-    """ Saves iptables firewall rules such they can be restored
+def block_iptable_rules_for_seconds(host, port_number, sleep_seconds, block_input=True, block_output=True):
+    """ For testing network partitions we alter iptables rules to block ports for some time.
+        We do that as a single SSH command because otherwise it makes it hard to ensure that iptable rules are restored.
     """
-
-    shakedown.run_command_on_agent(
-        host,
-        'if [ ! -e {} ] ; then sudo iptables-save > {} ; fi'.format(filename, filename))
-
-
-def restore_iptables(host, filename='iptables.rules'):
-    """ Reconnect a previously partitioned node to the network
-        :param hostname: host or IP of the machine to partition from the cluster
-    """
-
-    shakedown.run_command_on_agent(
-        host,
-        'if [ -e {} ]; then sudo iptables-restore < {} && sudo rm {} ; fi'.format(filename, filename, filename))
-
-
-@contextlib.contextmanager
-def iptable_rules(host):
     filename = 'iptables-{}.rules'.format(uuid.uuid4().hex)
-    save_iptables(host, filename)
-    try:
-        yield
-    finally:
-        # return config to previous state
-        restore_iptables(host, filename)
+    cmd = """
+          if [ ! -e {backup} ] ; then sudo iptables-save > {backup} ; fi;
+          {block}
+          sleep {seconds};
+          if [ -e {backup} ]; then sudo iptables-restore < {backup} && sudo rm {backup} ; fi
+        """.format(backup=filename, seconds=sleep_seconds,
+                   block=iptables_block_string(block_input, block_output, port_number))
+
+    shakedown.run_command_on_agent(host, cmd)
```

Comment:
Isn't this a blocking call?

Reply:
I think it is... I think right now it is not a problem. If we move to parallel testing than it might be problem...